README.md

```diff
@@ -91,97 +106,181 @@ cloning over SSH may provide a better experience (which requires
 
-    brew install cmake ninja
+If you need to work with other SDKs, you'll need to create a second build using
+Ninja.
 
-**[MacPorts](https://macports.org)**
+To build using Ninja, run:
 
-    sudo port install cmake ninja
+    utils/build-script --release
 
-### Building Swift
+If you are interested in compiler development it helps to build LLVM + Clang
+with optimizations enabled, but enable debug support in Swift. This is a good
+starting point since the debugger will work in the swift code base, and the
+tests will run faster.
 
-The `build-script` is a high-level build automation script that supports basic
-options such as building a Swift-compatible LLDB, building the Swift Package
-Manager, building for iOS, running tests after builds, and more. It also
-supports presets, which you can define for common combinations of build options.
+    utils/build-script --release-debuginfo --debug-swift
 
-To find out more:
+If you are interested in standard library development, this will build a fully
+optimized compiler, but will build the standard library and swift overlays with
+debug information.
 
-    utils/build-script -h
+    utils/build-script --release --debug-swift-stdlib
 
-Note: Arguments after "--" above are forwarded to `build-script-impl`, which is
-the ultimate shell script that invokes the actual build and test commands.
 
-A basic command to build Swift with optimizations and run basic tests with
-Ninja:
+#### Build Products
 
-    utils/build-script -r -t
+All of the build products are placed in `swift-source/build/${TOOL}-${MODE}/${PRODUCT}-${PLATFORM}/`.
+If macOS swift with Ninja in DebugAssert mode was built, all of the products
+would be in `swift-source/build/Ninja-DebugAssert/swift-macosx-x86_64/`. It
+helps to save this directory as an environment variable for future use.
 
-## Developing Swift in Xcode
+    export SWIFT_BUILD_DIR="~/swift-source/build/Ninja-DebugAssert/swift-macosx-x86_64"
 
-`build-script` can also generate Xcode projects:
+#### Ninja
 
-    utils/build-script -x
+Once the first build has completed, ninja can perform fast incremental builds of
+various products. These incremental builds are a big timesaver when developing
+and debugging.
 
-The Xcode IDE can be used to edit the Swift source code, but it is not currently
-fully supported as a build environment for SDKs other than macOS. If you need to
-work with other SDKs, you'll need to create a second build using Ninja.
+    cd ${SWIFT_BUILD_DIR}
+    ninja swift
+
+This will build the swift compiler, but will not rebuild the standard library or
+any other target. Building the `swift-stdlib` target as an additional layer of
+testing from time to time is also a good idea.
+
+    ninja swift-stdlib
+
+This command will rebuild the stdlib and skip the `benchmarks` target. This
+is great for developers interested in working on the standard library.
+
+It is always a good idea to do a full build after using `update-checkout`.
```

Comment:
Question. Was the notion of a scheme used in update-checkout confusing to you? Do you think it needs to be explained?